marshal.go

```diff
@@ -126,8 +131,8 @@ func (i *InstanceMetadata) MarshalJSON() ([]byte, error) {
 func (i InstanceMetadata) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
 	tokens := []xml.Token{start}
 
-	if i.parsed != nil {
-		for key, value := range i.parsed {
+	if i.Parsed != nil {
```

Comment:
The same comment about checking the entry count applies here. But note that the `range` expression on line 135 would still work fine (that is, never executing the surrounding `for` expression's body) against a nil or empty map reference.